chroma_core/models/step_result.py

```diff
@@ -47,6 +52,14 @@ class StepResult(models.Model):
 
     _step_types = {}
 
+    # Assign the dependent fields, we cannot assign these fields in the constructor,
+    # therefore we intercept the save() method
+    def save(self, *args, **kwargs):
```

Comment:
should these check to see if they've already been set before overwriting?

Reply:
Could you reveal more the undesired situation that might happen?